src/anemoi/utils/mlflow/auth.py

```diff
@@ -124,17 +186,50 @@ def refresh_token(self, value: str) -> None:
         self._refresh_token = value
         self.refresh_expires = time.time() + (REFRESH_EXPIRE_DAYS * 86400)  # 86400 seconds in a day
 
+    @staticmethod
+    def _get_store() -> ServerStore:
+        """Read the server store from disk."""
+        with CONFIG_LOCK:
+            file = TokenAuth._config_file
+            path = config_path(file)
+
+            if not os.path.exists(path):
+                save_config(file, {})
+
+            if os.path.exists(path) and os.stat(path).st_mode & 0o777 != 0o600:
+                os.chmod(path, 0o600)
+
+            return ServerStore(**load_raw_config(file))
+
+    @staticmethod
+    def get_servers() -> list[tuple[str, int]]:
+        """List of all saved servers, as a tuple (url, refresh_expires). Ordered most recently used first."""
+        return TokenAuth._get_store().servers
+
     @staticmethod
     def load_config() -> dict:
-        path = config_path(TokenAuth.config_file)
+        """Load the last used server configuration
 
-        if not os.path.exists(path):
-            save_config(TokenAuth.config_file, {})
+        Returns
+        -------
+        config : dict
+            Dictionary with the following keys: `url`, `refresh_token`, `refresh_expires`.
+            If no configuration is found, an empty dictionary is returned.
+        """
+        warnings.warn(
+            "TokenAuth.load_config() is deprecated and will be removed in a future release.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+
+        store = TokenAuth._get_store()
 
-        if os.path.exists(path) and os.stat(path).st_mode & 0o777 != 0o600:
-            os.chmod(path, 0o600)
+        last = {}
+        for url, cfg in store.items():
```

Comment:
Gert- the refresh expires get updated every time some runs anemoi-training train? Mostly asking to check the logic for sorting won't conflict in the case of using the multi-server json for syncing where the two servers will be used at the same time

Reply:
> the refresh expires get updated every time some runs anemoi-training train? 

Yes, but only once at the end of the training run.

I don't think this will impact the syncing with two servers, because in the syncing code we won't use this `load_config` function, so it won't use this sorting code. In there, we will just instantiate 2 `TokenAuth` objects each with their own URL, and then they each get the token for that URL.